pkg/schedulingtypes/replicascheduler.go

```diff
@@ -273,26 +273,103 @@ func (s *ReplicaScheduler) GetSchedulingResult(rsp *fedschedulingv1a1.ReplicaSch
 		}
 	}
 
-	plnr := planner.NewPlanner(rsp)
+	plnr := planner.NewReplicaPlanner(rsp)
 	return schedule(plnr, key, clusterNames, currentReplicasPerCluster, estimatedCapacity), nil
 }
 
-func schedule(planner *planner.Planner, key string, clusterNames []string, currentReplicasPerCluster map[string]int64, estimatedCapacity map[string]int64) map[string]int64 {
+type ReplicaScheduleResult struct {
+	result map[string]int64
+}
+
+func NewReplicaScheduleResult(result map[string]int64) *ReplicaScheduleResult {
+	return &ReplicaScheduleResult{
+		result: result,
+	}
+}
+
+func (r *ReplicaScheduleResult) clusterNames() []string {
+	clusterNames := []string{}
+	for name := range r.result {
+		clusterNames = append(clusterNames, name)
+	}
+	return clusterNames
+}
+
+func (r *ReplicaScheduleResult) SetPlacementSpec(obj *unstructured.Unstructured) {
```

Comment:
I don't think this method and the one that follows should exist in a type-specific implementation.   Placement is common across all federated types.